src/mongo_logger.ts

```diff
@@ -701,6 +708,7 @@ export class MongoLogger {
   componentSeverities: Record<MongoLoggableComponent, SeverityLevel>;
   maxDocumentLength: number;
   logDestination: MongoDBLogWritable | Writable;
+  pendingLog: any | undefined;
```

Comment:
```suggestion
  pendingLog: unknown = null;
```

isPromiseLike will determine if this is nullish or not and narrow it to something that supports a .then method.

Reply:
Yes, the function can take `unknown` as input, it'll force a few more checks but we'll confirm any input is a promise.

Reply:
Sounds good, I updated the type